lib/src/observable_value.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+/// A single variable with a stream emitting the variable's changing values.
+///
+/// Contains a single [value] which can be read and written as any other
+/// variable.
+///
+/// Whenever the value is set, the new value is emitted on the
+/// [updates] stream.
+/// Further, any new listener on the [updates] stream is initially sent the
+/// current [value].
+///
+/// Consider using immutable types for [value] since listeners will not
+/// be notified if the internal state of [value] changes, only if
+/// the [value] variable is written to.
+///
+/// Example:
+/// ```dart
+/// class ToggleWidget {
+///   final ObservableValue<bool> _isVisible = ObservableValue(true);
+///
+///   // ...
+///
+///   Stream<bool> get visibilityChanges => _isVisible.updates;
+///
+///   void toggleVisibility() {
+///     _isVisible.value ^= true; // XOR with true toggles value.
+///   }
+/// }
+/// ```
+abstract class ObservableValue<T> {
```

Comment:
I'm not totally happy with the name.
It's valid, but I fear "Observable" might be overused and it might be mistaken for meaning something specific to another use of the word.

Alternatives welcome!

Maybe `ValueNotifier` (focuses on the notification, not the value, but the notification is what distinguishes it from a normal variable), `StateStream` (too vague), `ValuePublisher` (better, but fear it might be confusing to new readers - Nate, wdyt?).

Reply:
I would vote for `ValuePublisher`